app/controllers/trainees/subject_specialisms_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Trainees
+  class SubjectSpecialismsController < ApplicationController
+    before_action :authorize_trainee
+    before_action :load_subject_specialisms
+
+    def edit
+      @subject_specialisms_form = LanguageSubjectSpecialismsForm.new(trainee)
+    end
+
+    def update
+      @subject_specialisms_form = LanguageSubjectSpecialismsForm.new(trainee, params: subject_specialism_params, user: current_user)
+
+      save_strategy = trainee.draft? ? :save! : :stash
+
+      if @subject_specialisms_form.public_send(save_strategy)
+        # TODO: do something
```

Comment:
i.e. redirect to our confirm form (which will most likely be different to the one we currently have)